### ISSUE TITLE
Why are outputs of XLM non-deterministic for a given input?

### ISSUE THREAD

**Issue by Akella17** [facebookresearch/XLM#101]
@glample @aconneau Hey, I was trying to generate the sentence embeddings using the ```generate-embeddings.ipynb``` notebook. However, the outputs of the model are different for the same input on multiple re-runs. Where is this non-determinism even coming from in a pretrained model with no recurrent connection?

At first, I thought that some parameters must be missing in the pretrained model, which leads to random initialization and different outputs for each initialization. However, even for a single initialization of model parameters, I am getting different outptuts. For instance, modifying the last code snippet of ```generate-embeddings.ipynb``` as follows gives different outputs.
```Python
tensor = model('fwd', x=word_ids, lengths=lengths, langs=langs, causal=False).contiguous()
print(tensor[:2,0,:2])
tensor = model('fwd', x=word_ids, lengths=lengths, langs=langs, causal=False).contiguous()
print(tensor[:2,0,:2])
```
```
tensor([[ 4.6262,  0.9216],
        [ 0.7369, -1.7287]], grad_fn=<SliceBackward>)
tensor([[ 2.8505, -4.8434],
        [-0.6131, -4.0123]], grad_fn=<SliceBackward>)
```

**Comment by glample** [comment]
Can you try to call `model.eval()` first and see if the outputs are the same?

**Comment by Akella17** [comment]
Thanks for the help. Calling ```model.eval()``` fixes this issue. The reason for non-determinism must have been batch-norm and dropout layers.
